public/js/profile.js

```diff
@@ -0,0 +1,166 @@
+.then((doc) => {
+  console.log(doc.data());
+    let readRow = `
+    <div class="filebox">
+    <input class ="profile_image" type="image" placeholder="이미지를 넣어주세요." src="${doc.data().image}">
+    <span class="material-icons removeimg_btn">delete</span>
+    <label for="file">NEW IMAGE</label> 
+    <input type="file" id="file">
+    </div>
+    <div class="area--txt profile_area--txt">
+    <div class="profile-msg-box">
+    <h3>Name</h3>
+    <input type="text" class="profile-msg" id="profile-name" value="${doc.data().name}" placeholder="${doc.data().name}" readonly>
+    </div>
+    <div class="profile-msg-box">
+    <h3>E-mail</h3>
+    <input type="text" class="profile-msg" id="profile-email" value="${doc.data().email}" placeholder="${doc.data().email}">
+    </div>
+    <div class="profile-msg-box">
+    <h3>Phone number</h3>
+    <input type="text" class="profile-msg" id="profile-phone_number" value="${doc.data().phone_number}" placeholder="${doc.data().phone_number}">
+    </div>
+    <div class="profile-msg-box">
+    <h3>Rank</h3>
+    <input type="text" class="profile-msg" id="profile-rank" value="${doc.data().rank}" placeholder="${doc.data().rank}">
+    </div>
+    </div>
+    `;
+    //input에 기존 값 넣어놓기
+    let initialName = "${doc.data().name}";
+    $('input[id="profile-name"]').attr('value',initialName);
+    let initialEmail = "${doc.data().email}";
+    $('input[id="profile-email"]').attr('value',initialEmail);
+    let initialPhoneNumber = "${doc.data().phone_number}";
+    $('input[id="profile-phone_number"]').attr('value',initialPhoneNumber);
+    let initialRank = "${doc.data().rank}";
+    $('input[id="profile-rank"]').attr('value',initialRank);
+  $('.profile_area').append(readRow);
+});
+
+
+
+//---------------UPDATE---------------//
+
+//update 버튼
+$('.update_btn').click(function(){
+
+  console.log('update 버튼 누름');
+
+  //storage에 img 저장
+  let imageFile = document.querySelector('#file').files[0];
+  let saveWay =  ref(storage, 'image/' + imageFile.name);
+  uploadBytes(saveWay, imageFile).catch(failUpdate).then((snapshot) => {
+    //URL 따서, txt에 주기
+    getDownloadURL(snapshot.ref).then(async (url) => {
+    console.log(url);
+
+
+let updateItem = {
+  'name': $('#profile-name').val(),
+  'email': $('#profile-email').val(),
+  'phone_number': $('#profile-phone_number').val(),
+  'rank': $('#profile-rank').val(),
+  'image' : url
+};
+const updateProfile = await updateDoc(docRef, updateItem)
+.catch(failUpdate).then(successUpdate);
+})
+}
+)  
+});
+
+
+async function successUpdate() {
+  try {
+    await alert ('수정에 성공했습니다.');
+  } finally {
+    console.log('HOMESWEETHOME');
+    window.location.href='/index.html';
+  }
+};
+
+function failUpdate() {
+  alert ('수정에 실패했습니다. 새로고침 후 다시 시도해 주세요.')
+  console.log(err)
+};
+
+
+
+//profile 삭제
+$('.delete_btn').click(()=>{
+  console.log('del버튼 누름');
+  deleteDoc(doc(db, "product", nameId))
+  .catch(failDelete).then(successDelete);
+
+})
```

Comment:
```suggestion
$('.delete_btn').click(()=>{
  console.log('del버튼 누름');
  
  deleteDoc(doc(db, "product", nameId))
  .catch(failDelete)
  .then(successDelete);
})
```

.() 이런식으로 체이닝할 때는 줄바꿈하는 게 가독성이 좋아요. 그리고 적절한 엔터도 필요합니다!